homeassistant/components/mystrom/sensor.py

```diff
@@ -0,0 +1,95 @@
+"""Support for myStrom sensors of switches/plugs."""
+from __future__ import annotations
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import (
+    PLATFORM_SCHEMA,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_NAME, UnitOfPower, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, MANUFACTURER
+
+DEFAULT_NAME = "myStrom Switch"
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    }
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the myStrom entities."""
+    device = hass.data[DOMAIN][entry.entry_id].device
+    async_add_entities(
+        [
+            MyStromSwitchConsumptionSensor(device, entry.title),
+            MyStromSwitchTemperatureSensor(device, entry.title),
+        ]
+    )
+
+
+class MyStromSwitchConsumptionSensor(SensorEntity):
```

Comment:
Both sensors are having the same behavior, Take a look at using SensorEntityDescriptions. This way you have one base sensor where you inject different types of sensor types.

Reply:
that sounds very interesting but I was not able to find this in the doc, could you give me a hint how that works?

Reply:
Out the top of my head, checkout `homeassistant/components/youtube/sensor.py` to get an indication

Reply:
I see there are many sensors like that, I'll have time for this improvements somewhere in the next week